ingest/api/ingest.py

```diff
@@ -31,57 +32,67 @@ def __init__(
         self.uuid_prefix = uuid_prefix
 
         if not schema_path:
-            self.schema_path = pkg_resources.resource_filename("esoh", "schemas")
+            self.schema_path = pkg_resources.resource_filename("ingest", "schemas")
         else:
             self.schema_path = schema_path
         if not schema_file:
             self.schema_file = "e-soh-message-spec.json"
         else:
             self.schema_file = schema_file
-
         esoh_mqtt_schema = os.path.join(self.schema_path, self.schema_file)
+
         with open(esoh_mqtt_schema, "r") as file:
             self.esoh_mqtt_schema = json.load(file)
         self.schema_validator = Draft202012Validator(self.esoh_mqtt_schema)
 
         if testing:
             return
 
-        self.dstore = DatastoreConnection(dstore_conn["dshost"], dstore_conn["dsport"])
+        # self.dstore = DatastoreConnection(dstore_conn["dshost"], dstore_conn["dsport"])
         if "username" in mqtt_conf:
             self.mqtt = MQTTConnection(
                 mqtt_conf["host"], mqtt_conf["topic"], mqtt_conf["username"], mqtt_conf["password"]
             )
         else:
             self.mqtt = MQTTConnection(mqtt_conf["host"], mqtt_conf["topic"])
 
-    def ingest(self, message: [str, object], input_type: str = None):
+    def ingest(self, message: Union[str, object], input_type: str = None):
         """
         Method designed to be main interaction point with this package.
         Will interpret call all methods for deciding input type, build the mqtt messages, and
         publish them.
 
         """
         if not input_type:
-            input_type = self.decide_input_type(message)
-
-        self.publish_messages(self._build_messages(message, input_type))
+            input_type = self._decide_input_type(message)
+        messages = self._build_messages(message, input_type)
+        if isinstance(messages, list):
+            response, status = self.publish_messages(messages)
+            return response, status
+        else:
+            return messages, 400
 
     def publish_messages(self, messages: list):
         """
         This method accepts a list of json strings ready to be ingest to datastore
          and published to the mqtt topic.
         """
+        if len(messages) > 0:
+            topic = messages[0]["properties"]["naming_authority"]
         for msg in messages:
             if msg:
                 try:
-                    self.dstore.ingest(msg)
-                    self.mqtt.send_message(msg)
-                except grpc.RpcError:
-                    self.dstore.is_channel_ready()
-                    pass
-                except Exception:
-                    pass
+                    ingest(msg)
+                except grpc.RpcError as v_error:
+                    return "Failed to ingest" + "\n" + str(v_error), 500
```

Comment:
the grpc error might contain internal information and probably should not always be returned to the user. There is no need to inform of how data is passed to the datastore, but we should tell the user about what they could do to resolve the error. I.e "API could not reach datastore" is one option, and the user should know they did nothing wrong.